lighthouse-core/gather/fetcher.js

```diff
@@ -6,124 +6,56 @@
 'use strict';
 
 /**
- * @fileoverview Fetcher is a utility for making requests within the context of the page.
- * Requests can circumvent CORS, and so are good for fetching source maps that may be hosted
- * on a different origin.
+ * @fileoverview Fetcher is a utility for making requests to any arbitrary resource,
+ * ignoring normal browser constraints such as CORS.
  */
 
-/* global document */
+/* global fetch */
 
 /** @typedef {{content: string|null, status: number|null}} FetchResponse */
 
-const log = require('lighthouse-logger');
-const {getBrowserVersion} = require('./driver/environment.js');
-
 class Fetcher {
```

Comment:
We don't need to use a class anymore. We could refactor this to be more function like the rest of our helper files in `lighthouse-core/gather/driver`. This way we don't need to pass around a `fetcher` object on the gather context.

Reply:
I think we should do this in a different PR.